gradle/ci-support.gradle

```
@@ -0,0 +1,33 @@
+import groovy.json.JsonOutput
+
+// Emit a JSON-formatted list of check tasks to be run in CI
+task testMatrix {
+    project.afterEvaluate {
+        def checkTasks = subprojects.collect {
+            it.tasks.findByName("check")
+        }.findAll { it != null }
+
+        dependsOn(checkTasks)
+        doLast {
+            def checkTaskPaths = checkTasks
+                .findAll { !it.state.upToDate }
+                .collect { it.path }
+
+            println(JsonOutput.toJson(checkTaskPaths))
+        }
+    }
+}
+
+gradle.taskGraph.whenReady {
```

Worth adding a comment to explain this?
```suggestion
// If we're executing the `taskMatrix` task, disable tests and other slow tasks
//  so that we can get a result quickly.
gradle.taskGraph.whenReady {
```